OKR.md

```diff
@@ -1,49 +1,64 @@
+
+### Key results
+**[IPFS Companion](https://github.com/ipfs/ipfs-companion) is the reference implementation of IPFS in the browser**
+  - Simplified and improved UX  ([ipfs-companion/#324](https://github.com/ipfs-shipyard/ipfs-companion/issues/324) [ipfs-companion/#342](https://github.com/ipfs/ipfs-companion/issues/342))
+  - Welcoming, up-to-date developer documentation exists
+  - Retain positive presence in browser extension stores
+  - Robust CI/QA practices are established, improve test coverage ([ipfs-companion/#145](https://github.com/ipfs/ipfs-companion/issues/145))
+  - Detect [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop) and provide additional controls ([ipfs-companion/#350](https://github.com/ipfs-shipyard/ipfs-companion/issues/350))
+  - Embedded `js-ipfs` node used when a system ipfs daemon is not availble.
+  - IPFS is embeded into every browser page through `window.ipfs` ([ipfs-compaion/#330](https://github.com/ipfs-shipyard/ipfs-companion/issues/330))
+    - Opt-in or access controls exist to prevent web pages from performing actions without prior user approval
+    - `window.ipfs` is a proxy over `window.postMessage` to communicate with a running IPFS node exposed by the extension ([ipfs-postmsg-proxy](https://github.com/tableflip/ipfs-postmsg-proxy))
+  - Update PeerPad to use the `window.ipfs` if it is available
+  - Provide a step by step guide for devs to create apps that use `window.ipfs`, using PeerPad as the example.
+
+**Get the [Brave](https://brave.com) integration released**
+
+- IPFS companion working out-of-the-box in Brave
+- Support for `ipfs://` and `dweb:/ipfs` address schemes in Location Bar and DOM elements (`href`, `src`)
+
+**IPFS Desktop replaces go-ipfs as the recommended starting point for new users**
+
+  - TBC (what are the plans for ipfs-desktop ?)
+  - Update _"getting started"_ section of https://ipfs.io
+  - `ipfs-dashboard` is built as a reusable replacement for `ipfs-web-ui`
+  - Update `ipfs-dashboard` to use `window.ipfs` if available
+  - Create a ui-style-guide repo as a guide to the design language of IPFS apps, and share look and feel with `ipfs-companion`, `ipfs-desktop` and `ipfs-dashboard`
+  - A logo for dashboard, desktop and companion to give each an identity that shows they separate parts that work together?
+
+## Understand how IPFS can be integrated into web browsers
+
+_Support the conversation with browser vendors. We need to make it as easy as possible for them to see where we are at, what's needed and why._
+
+### Key results
+
+**Set up `arewedistributedyet.com` as a communal call to action.** ([in-web-browsers/#24](https://github.com/ipfs/in-web-browsers/issues/24))
+- Publish the list of apis needed, with a tests where possible, a short summary on what it would allow, and a list of p2p projects that want it.    
+
+**Create specs for addressing on the Decentralized Web**
```

Comment:
Start thinking about how you will make this one measurable. It should be worded in a way that it is unambiguous what it means to get 0.6 on this KR or 1.0.

Reply:
There is an outline at https://github.com/ipfs/specs/tree/master/dweb-addressing with a minimum set of items we want to cover in these specs.
At this point it makes sense to keep it all as a single document, unless it turns out beneficial to split it.

Reply:
That outline covers ipfs:// and ipns:// URL schemes. As you've heard me comment, ipns:// URLs have a problem with link rot. The really should be deprecated in favor of an ipfs:// URL to an IPLD object which is a snapshot the current IPNS record (at the time of creating the URL) plus the peer ID for lookup up the latest IPNS record (which can fail if the original publisher stops posting updates). Also, the existing ipns:// URL (which is a public key hash) is not secure in the long term since, years later, the private key may be compromised and the URL is untrustworthy. (An ipfs:// URL to an IPNS snapshot does not have this vulnerability.)

Reply:
@jefft0 noted (https://github.com/ipfs/specs/issues/166#issuecomment-356743980 is a better place for this discussion), will get back to it when we start work on dweb addressing spec.  👍 

Back to this PR, I think we should keep this OKR short, perhaps just skip protocol names altogether and replace KR with "addressing schemes are documented" + reference to outline doc.

Reply:
just make sure you have some document that defines, at the beginning of the quarter, what the goal is and how you will measure it. ie. if you add 10 more sections to the outline of the dweb-addressing document next month, are you aiming to have the _originally outlined_ sections complete or are you aiming to have a _final draft_ of the full document? Are you aiming to have the document completed, ratified, and merged into ipfs/specs with a corresponding blog post?